src/main/java/net/neoforged/neoforge/registries/DeferredItem.java

```diff
@@ -9,14 +9,34 @@
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.ItemLike;
 
 /**
  * Special {@link DeferredHolder} for {@link Item Items} that implements {@link ItemLike}.
- * 
+ *
  * @param <T> The specific {@link Item} type.
  */
 public class DeferredItem<T extends Item> extends DeferredHolder<Item, T> implements ItemLike {
+    /**
+     * Creates a new {@link ItemStack} with a default size of 1 from this {@link Item}
+     */
+    public ItemStack toStack() {
+        return toStack(1);
+    }
+
+    /**
+     * Creates a new {@link ItemStack} with the given size from this {@link Item}
+     *
+     * @param count The size of the stack to create
+     */
+    public ItemStack toStack(int count) {
+        ItemStack stack = asItem().getDefaultInstance();
+        if (stack.isEmpty()) throw new IllegalStateException("Item stack can't be null!");
```

Comment:
The error message is misleading, since the item stack isn't null (item stacks normally are never null). The error message should be improved; perhaps "Obtained empty item stack; incorrect getDefaultInstance() call?" (because the only time this check should ever succeed is if `getDefaultInstance()` returns an empty stack, by virtue of a subclass's override.

It might also be worth considering removing the check entirely, since the above is plain and very obvious developer error which might very rarely if not never pop up.

Reply:
I think having this here too makes it more consistent with `DeferredBlock`, correct me if I am wrong.

Reply:
Hmmm. Sure, we can leave it there for consistency.